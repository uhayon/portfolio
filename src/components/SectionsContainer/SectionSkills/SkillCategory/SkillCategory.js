import React from 'react';
import PropTypes from 'prop-types';

import styles from './SkillCategory.module.scss';

const SkillCategory = ({ selectedLanguage, categoryName, category }) => {
  return (
    <div className={styles.skillCategory}>
      <div className={styles.skillCategoryHeader}>
        <h3>{category.title[selectedLanguage]}</h3>
        <img height='100px' src={require(`../../../../assets/icon-${categoryName}.svg`)} alt={categoryName} />
      </div>
      <div style={{padding: '1em'}}>{category.skills.join(', ')}</div>
    </div>
  )
};

SkillCategory.propTypes = {
  selectedLanguage: PropTypes.string.isRequired,
  categoryName: PropTypes.string.isRequired,
  category: PropTypes.object.isRequired
}

export default SkillCategory;