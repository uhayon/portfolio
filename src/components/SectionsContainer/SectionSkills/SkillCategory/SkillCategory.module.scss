@import '../../../../globalStyles/colors.scss';

.skillCategory {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 300px;
    height: 500px;
    background-color: rgba(255, 255, 255, .8);
    margin: 1em;
    text-align: center;
    border-radius: 10px;
    transform: scale(1);
    transition: all 0.2s linear;
    &:hover {
        border: 2px solid $red;
        transform: scale(1.1);
        font-weight: bold;
        background-color: $gray-dark;
        color: $white;
        font-size: 1.1rem;
    }

    .skillCategoryHeader {
        width: 100%;
    }
}