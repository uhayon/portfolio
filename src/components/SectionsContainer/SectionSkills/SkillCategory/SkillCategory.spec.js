import React from 'react';
import { shallow } from 'enzyme';

import SkillCategory from './SkillCategory';

describe('Mounting Skill Category', () => {
  it('Should match the snapshot', () => {
    const category = {
      title: {
        EN: 'title'
      },
      skills: []
    };

    const wrapper = shallow(<SkillCategory category={category} categoryName='tools' selectedLanguage='EN' />);
    expect(wrapper).toMatchSnapshot();
  });
});