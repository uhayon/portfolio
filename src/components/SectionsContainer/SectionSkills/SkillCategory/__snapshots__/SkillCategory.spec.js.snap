// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounting Skill Category Should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SkillCategory
    category={
      Object {
        "skills": Array [],
        "title": Object {
          "EN": "title",
        },
      }
    }
    categoryName="tools"
    selectedLanguage="EN"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="skillCategoryHeader"
        >
          <h3>
            title
          </h3>
          <img
            alt="tools"
            height="100px"
            src={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "icon-tools.svg",
              }
            }
          />
        </div>,
        <div />,
      ],
      "className": "skillCategory",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              title
            </h3>,
            <img
              alt="tools"
              height="100px"
              src={
                Object {
                  "ReactComponent": [Function],
                  "__esModule": true,
                  "default": "icon-tools.svg",
                }
              }
            />,
          ],
          "className": "skillCategoryHeader",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "title",
            },
            "ref": null,
            "rendered": "title",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "tools",
              "height": "100px",
              "src": Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "icon-tools.svg",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="skillCategoryHeader"
          >
            <h3>
              title
            </h3>
            <img
              alt="tools"
              height="100px"
              src={
                Object {
                  "ReactComponent": [Function],
                  "__esModule": true,
                  "default": "icon-tools.svg",
                }
              }
            />
          </div>,
          <div />,
        ],
        "className": "skillCategory",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                title
              </h3>,
              <img
                alt="tools"
                height="100px"
                src={
                  Object {
                    "ReactComponent": [Function],
                    "__esModule": true,
                    "default": "icon-tools.svg",
                  }
                }
              />,
            ],
            "className": "skillCategoryHeader",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "title",
              },
              "ref": null,
              "rendered": "title",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "tools",
                "height": "100px",
                "src": Object {
                  "ReactComponent": [Function],
                  "__esModule": true,
                  "default": "icon-tools.svg",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
